subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpResourceAccessor.java

```diff
@@ -61,28 +59,20 @@ public HttpResponseResource openResource(final URI uri) {
      * Same as #getResource except that it always gives access to the response body,
      * irrespective of the returned HTTP status code. Never returns {@code null}.
      */
-    public HttpResponseResource getRawResource(final URI uri) {
+    public HttpResponseResource getRawResource(final URI uri, boolean revalidate) {
         abortOpenResources();
         String location = uri.toString();
         LOGGER.debug("Constructing external resource: {}", location);
-
-        HttpRequestBase request = new HttpGet(uri);
-        HttpResponse response;
-        try {
-            response = http.performHttpRequest(request);
-        } catch (IOException e) {
-            throw new HttpRequestException(String.format("Could not %s '%s'.", request.getMethod(), request.getURI()), e);
-        }
-
+        HttpResponse response = http.performGet(location, revalidate);
```

Comment:
Can we implement this change while still throwing the HttpRequestExeception when the resource is not accessible?  Note that when the resource is unaccessible with this change, the response now ends up being null and we throw an NPE from the wrapResponse() call on the next line.  So instead of bubbling up the HttpRequestExeception, we bubble up an NPE and the user gets a more cryptic error.  See https://github.com/gradle/gradle/blob/242799c43476c818176740fc202e7e8054cb09ed/subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/use/DeployedPortalIntegrationSpec.groovy#L94-L94 for a test that exemplifies this.